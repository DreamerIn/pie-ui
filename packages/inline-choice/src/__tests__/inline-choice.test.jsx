import React from 'react';
import InlineChoice from '../inline-choice';
import { shallow } from 'enzyme';

const props = {
  session: {},
  onChoiceChanged: () => {}
};

const model = {
  choices: [
    {
      correct: true,
      value: 'sweden',
      label: 'sweden',
      feedback: {}
    },
    {
      value: 'iceland',
      label: 'iceland',
      feedback: {
        type: 'default',
        text: [
          {
            lang: 'en-US',
            value: 'Iceland is not an EU member state'
          },
          {
            lang: 'es-ES',
            value: 'Islandia no es un estado miembro de la UE'
          },
          {
            lang: 'zh-CN',
            value: '冰岛不是欧盟成员国'
          }
        ]
      }
    }
  ],
  classes: {},
  disabled: false
};

describe('inline-choice', () => {
  describe('snapshot', () => {
    it('should render component', () => {
      const wrapper = shallow(
        <InlineChoice
          {...model}
          session={props.session}
          onChoiceChanged={props.onChoiceChanged}
        />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('logic', () => {
    xit('todo');
  });
});
