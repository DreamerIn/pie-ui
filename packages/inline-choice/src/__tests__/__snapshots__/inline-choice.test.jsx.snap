// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inline-choice snapshot should render component 1`] = `
<InlineChoice
  choices={
    Array [
      Object {
        "correct": true,
        "feedback": Object {},
        "label": "sweden",
        "value": "sweden",
      },
      Object {
        "feedback": Object {
          "text": Array [
            Object {
              "lang": "en-US",
              "value": "Iceland is not an EU member state",
            },
            Object {
              "lang": "es-ES",
              "value": "Islandia no es un estado miembro de la UE",
            },
            Object {
              "lang": "zh-CN",
              "value": "冰岛不是欧盟成员国",
            },
          ],
          "type": "default",
        },
        "label": "iceland",
        "value": "iceland",
      },
    ]
  }
  classes={
    Object {
      "container": "InlineChoice-container-1",
      "flex": "InlineChoice-flex-2",
      "formControl": "InlineChoice-formControl-3",
      "selectEmpty": "InlineChoice-selectEmpty-4",
    }
  }
  disabled={false}
  onChoiceChanged={[Function]}
  session={Object {}}
/>
`;
